packages/amplication-client/src/Workspaces/WorkspaceHeader/WorkspaceHeader.tsx

```diff
@@ -82,6 +90,71 @@ const WorkspaceHeader: React.FC = () => {
 
   const [versionAlert, setVersionAlert] = useState(false);
 
+  const [upgradeButtonData, setUpgradeButtonData] = useState<{
+    trialDaysLeft?: number;
+    trialLeftProgress?: number;
+    showUpgradeTrialButton: boolean;
+    showUpgradeDefaultButton: boolean;
+  }>({ showUpgradeTrialButton: false, showUpgradeDefaultButton: true });
+
+  const getCurrentSubscription = useCallback(async () => {
+    if (currentWorkspace) {
+      await stigg.setCustomerId(currentWorkspace.id);
+      const [subscription] = await stigg.getActiveSubscriptions();
+
+      if (!subscription) {
+        console.error("No active subscription found");
+        return;
+      }
+
+      const DAYS_TO_SHOW_VERSION_ALERT_SINCE_END_OF_TRIAL = 14;
+
+      if (
+        subscription.status === SubscriptionStatus.Active &&
+        subscription.plan.pricingType !== PricingType.Free
+      ) {
+        setUpgradeButtonData({
+          showUpgradeDefaultButton: false,
+          showUpgradeTrialButton: false,
+        });
+      } else if (subscription.plan.id === BillingPlan.Free) {
+        const daysSinceStartOfPlan = Math.abs(
+          (Date.now() - new Date(subscription.startDate).getTime()) / ONE_DAY
+        );
+
+        const customer = await stigg.getCustomer();
+        const previouslyOnTrial = customer.trialedPlans.length > 0;
+        const showUpgradeTrialButton =
+          previouslyOnTrial &&
+          daysSinceStartOfPlan < DAYS_TO_SHOW_VERSION_ALERT_SINCE_END_OF_TRIAL;
+
+        setUpgradeButtonData({
+          trialDaysLeft: 0,
+          trialLeftProgress: 0,
+          showUpgradeTrialButton,
+          showUpgradeDefaultButton: false,
+        });
+      } else {
+        const trialDaysLeft = Math.round(
+          Math.abs((subscription.trialEndDate.getTime() - Date.now()) / ONE_DAY)
+        );
+        const trialLeftProgress =
+          (100 * trialDaysLeft) / subscription.plan.defaultTrialConfig.duration;
+
+        setUpgradeButtonData({
+          trialDaysLeft,
+          trialLeftProgress,
+          showUpgradeTrialButton: true,
+          showUpgradeDefaultButton: false,
+        });
+      }
+    }
+  }, [currentWorkspace]);
+
+  useEffect(() => {
```

Comment:
this will run every rerender of the header. this is the reason I told you to pass it to the stigg hook and pass it inside the context. by the way, you wont need the state for that because it will happen once